### ISSUE TITLE
Support d3-style member expression indentation

### ISSUE THREAD

**Issue by pokey** [eslint/eslint#10538]
**What rule do you want to change?**

`indent`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option

**Please provide some example code that this change will affect:**

```js
d3.select("body")
  .append("svg")
    .attr("width", 960)
    .attr("height", 500)
  .append("g")
    .attr("transform", "translate(20,20)")
  .append("rect")
    .attr("width", 920)
    .attr("height", 460);
```

**What does the rule currently do for this code?**

`2:1   error  Expected indentation of 4 spaces but found 2`

**What will the rule do after it's changed?**

I'd want some way to support different indentation levels depending on the member name, as recommended for [`d3-select`](https://github.com/d3/d3-selection).  One possibility would be to have `MemberExpression` accept a list where the first element is the default indentation multiple and the second is an object that specifies different indentation multiples for different member names.  Something like this:

```js
["error", 4, { "MemberExpression": [1, {"append": 0.5}] }]
```

There's other possibilities, but I think this gives an idea of the flavour of rule that would be helpful.

**Comment by platinumazure** [comment]
This seems like it would basically be library-specific, which we don't usually like to do with core rules.

I think your best bet with the core rule is to use the `ignoredNodes` option to pass in selectors that represent the nodes with deeper indentations (either by specifying property names, or some other method). Then you could optionally write a custom d3 indent rule (maybe in a d3 ESLint plugin) that applies the library-specific style conventions.

**Comment by pokey** [comment]
Ah ok makes sense to me.  Thank you
